### DISCUSSION TITLE
Best practice for BERTopic.merge_models

### DISCUSSION THREAD

**Discussion by Matagi1996** [MaartenGr/BERTopic#1705] [Q&A]
As I understand it, this function merges the topics of several trained models, but not their Cluster (HDBSCAN) or PCA model parameters.
As such, am I right to assume it is not recommended to split a large dataset in several batches (models) and combine all of them later (kind of like a makeshift online mode) if want the to predict topics based on all datapoints?

**Comment by MaartenGr** [comment]
`.merge_models` was actually created, in part, for that specific feature. You can find a guide for incremental learning with that feature [here](https://maartengrootendorst.substack.com/i/139691971/model-merging-federated-and-incremental-learning). I would definitely advise using it for incremental learning. 

Even though it does not save cluster and dimensionality reduction parameters, it can still be used for incremental learning if the data chunks are large enough. The only thing to note is that the CountVectorizer is not merged (due to potentially different tokenization schemes).

**Comment by Matagi1996** [comment]
Thanks, I noticed today while looking into the code that when I safe with savetensor thouse parameters are not safed anyway so I might have worried to much.
 I updated to the new version to test merge and am getting 
 BERTopic - WARNING: You are loading a BERTopic model without explicitly defining an embedding model
 
 I pass this embeddingmodel what is simply a convinience wrapper around Sentencetransformer MiniLM-L12-v2 to push it to GPU and increese the embeding size a little bit, with the previouse version that worked well but now I always get the warning.
 
BERTopic.load(path_to_file, embedding_model=Embedder())

```
class Embedder(BaseEmbedder):
      def __init__(self, device:str="cuda",max_len:int=256):
          super().__init__()
          self.embedding_model =en_emb(device,max_len)
  
      def embed(self, documents, verbose=False):
          embeddings = self.embedding_model.encode(documents,normalize_embeddings=True)
          return embeddings


 def en_emb(device:str="cuda",max_len=256):
    cache = os.environ['TRANSFORMERS_CACHE']
    id = 'sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2'
    model =SentenceTransformer(model_name_or_path=id, device = device, cache_folder= cache)
    model.max_seq_length=max_len
    return model
```

Not critical but just I am just wondering what changed.

**Reply by MaartenGr** [reply]
It's difficult to say without seeing an end-to-end example (some code is missing in your example) but my guess would be that since there was no embedding model saved, it cannot overwrite it:

https://github.com/MaartenGr/BERTopic/blob/1285f54a8c034b30dbac8af83b550203c9d68240/bertopic/_bertopic.py#L3060

This might be a too restrictive statement. Having said that, you can also just run the following instead:

```python
from bertopic._bertopic import select_backend
topic_model.embedding_model = select_backend(Embedder())
```
